packages/payment-processor/test/payment/erc-20-private-payment-hinkal.test.ts

```diff
@@ -0,0 +1,256 @@
+        value: payeeAddress,
+      },
+      payer: {
+        type: Types.Identity.TYPE.ETHEREUM_ADDRESS,
+        value: payerWallet.address,
+      },
+      expectedAmount: currencyAmount.toString(),
+      timestamp: Utils.getCurrentTimestampInSecond(),
+    },
+    signer: {
+      type: Types.Identity.TYPE.ETHEREUM_ADDRESS,
+      value: payerWallet.address,
+    },
+    paymentNetwork,
+  };
+
+  // Step 5: create & send request
+  const request = await requestClient.createRequest(requestCreateParameters);
+  const requestData = await request.waitForConfirmation();
+  return requestData;
+};
+
+// This function is required to wait for the indexing of shielded balance changes in Hinkal
+const waitLittle = (time = 15): Promise<number> =>
+  new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(1);
+    }, time * 1000);
+  });
+
+// Helper function: allows to get a shielded balance of a specific token
+const getTokenShieldedBalance = async (
+  address: string,
+  tokenAddress = currencyAddress,
+): Promise<bigint> => {
+  const balances = await hinkalStore[address].getBalances();
+  const tokenBalance = balances.find(
+    (balance) => balance.token.erc20TokenAddress === tokenAddress,
+  )?.balance;
+
+  if (tokenBalance === undefined) {
+    throw Error('Shiedled Balance calculated incorrectly');
+  }
+  return tokenBalance;
+};
+
+describe('ERC-20 Private Payments With Hinkal', () => {
+  let provider: providers.Provider;
+  let payerWallet: ethers.Wallet;
+  let payeeWallet: ethers.Wallet;
+  beforeAll(async () => {
+    provider = new ethers.providers.JsonRpcProvider(RPC_URL);
+    payerWallet = Wallet.fromMnemonic(payerMnemonic).connect(provider);
+    payeeWallet = new Wallet(payeePrivateKey, provider);
+    await addToHinkalStore(payerWallet);
+  });
+  afterAll(async () => {
+    for (const key in hinkalStore) {
+      hinkalStore[key].snapshotsClearInterval(); // Clear Hinkal's snapshots interval: needed for jest to finish as a process
+    }
+    jest.clearAllTimers(); // Clear any pending Jest timers
+  });
+
+  describe('Privacy of a Sender: From Private to Public Transactions', () => {
+    // Objectives of this test:
+    // 1. The payer's address should never appear on-chain.
+    // 2. The payee should successfully receive the funds.
+    // 3. The transaction should complete successfully.
+
+    it('ERC-20 Proxy: Payer is not the same as Origin/Sender of Transaction', async () => {
+      const requestData = await createRequestForHinkal(
+        payerWallet,
+        Types.Extension.PAYMENT_NETWORK_ID.ERC20_PROXY_CONTRACT,
+      );
+      const balanceErc20Before = await getErc20Balance(requestData, payeeAddress, provider);
+
+      const tx = await payErc20ProxyRequestFromHinkalShieldedAddress(requestData, payerWallet);
+      const balanceErc20After = await getErc20Balance(requestData, payeeAddress, provider);
+
+      await waitLittle(5); // wait before balance is increased
```

Comment:
Don't we need to wait _before_ retrieving `balanceErc20After`? Or is there no need to wait if the test is already passing?

Reply:
I'll try removing the wait.